test/test_common/environment.cc

```diff
@@ -342,8 +342,8 @@ std::string TestEnvironment::temporaryFileSubstitute(const std::string& path,
 }
 
 std::string TestEnvironment::readFileToStringForTest(const std::string& filename,
-                                                     bool require_existence) {
-  std::ifstream file(filename, std::ios::binary);
+                                                     bool require_existence, bool read_binary) {
```

Comment:
I'm a little worried about this proliferating in options; should we have a `struct` for options here? We can keep the existing signature for backwards compatibility as a wrapper to avoid changing the entire world.

Reply:
I can't envision additional flags, so a struct seems like overkill.

Reply:
Arguably it makes the tests that have been modified easier to read (should the default be true or false?), but I don't feel that strongly.

Reply:
I'm personally a fan of bitwise flags, and was annoyed as we wrote this that named args aren't a c++ thing, so I'll rewrite it with enums if you insist. But I'd like a use case that suggests a third toggle before we bother.